pkg/tsdb/cloudwatch/models/request_parser_test.go

```diff
@@ -124,255 +138,486 @@ func TestRequestParser(t *testing.T) {
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: MetricEditorModeBuilder,
+			expectedGMDApiMode:       GMDApiModeMetricStat,
+		},
+		"no metric query type, no metric editor mode, has expression": {
+			extraDataQueryJson:       `"expression":"SUM(a)",`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: MetricEditorModeRaw,
+			expectedGMDApiMode:       GMDApiModeMathExpression,
+		},
+		"no metric query type, has metric editor mode, has expression": {
+			extraDataQueryJson:       `"expression":"SUM(a)","metricEditorMode":99,`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: dummyTestEditorMode,
+			expectedGMDApiMode:       GMDApiModeMetricStat,
+		},
+		"no metric query type, has metric editor mode, no expression": {
+			extraDataQueryJson:       `"metricEditorMode":99,`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: dummyTestEditorMode,
+			expectedGMDApiMode:       GMDApiModeMetricStat,
+		},
+		"has metric query type, has metric editor mode, no expression": {
+			extraDataQueryJson:       `"type":"timeSeriesQuery","metricEditorMode":99,`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: dummyTestEditorMode,
+			expectedGMDApiMode:       GMDApiModeMetricStat,
+		},
+		"has metric query type, no metric editor mode, has expression": {
+			extraDataQueryJson:       `"type":"timeSeriesQuery","expression":"SUM(a)",`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: MetricEditorModeRaw,
+			expectedGMDApiMode:       GMDApiModeMathExpression,
+		},
+		"has metric query type, has metric editor mode, has expression": {
+			extraDataQueryJson:       `"type":"timeSeriesQuery","metricEditorMode":99,"expression":"SUM(a)",`,
+			expectedMetricQueryType:  MetricQueryTypeSearch,
+			expectedMetricEditorMode: dummyTestEditorMode,
+			expectedGMDApiMode:       GMDApiModeMetricStat,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			query := []backend.DataQuery{
+				{
+					JSON: json.RawMessage(fmt.Sprintf(
+						`{
+						   "refId":"ref1",
+						   "region":"us-east-1",
+						   "namespace":"ec2",
+						   "metricName":"CPUUtilization",
+						   "statistic":"Average",
+							%s
+						   "period":"900"
+						}`, tc.extraDataQueryJson),
+					),
+				},
+			}
+			res, err := ParseMetricDataQueries(query, time.Now(), time.Now(), false)
 			require.NoError(t, err)
-			require.True(t, res.ReturnData)
+			require.Len(t, res, 1)
+			require.NotNil(t, res[0])
+			assert.Equal(t, tc.expectedMetricQueryType, res[0].MetricQueryType)
+			assert.Equal(t, tc.expectedMetricEditorMode, res[0].MetricEditorMode)
+			assert.Equal(t, tc.expectedGMDApiMode, res[0].GetGMDAPIMode())
 		})
-	})
+	}
+}
 
-	t.Run("ID is the string `query` appended with refId if refId is a valid MetricData ID", func(t *testing.T) {
```

Comment:
tests were shifted down to around line 552 https://github.com/grafana/grafana/pull/57431/files#diff-e2107889a319fc8692d34cb7ed8146ba186329754d1dcd33324ba4c29862713bR552